vyper/builtins/functions.py

```diff
@@ -2575,10 +2579,13 @@ def evaluate(self, node):
         val = self._eval(input_type)
 
         if isinstance(input_type, DecimalT):
-            return vy_ast.Decimal.from_node(node, value=val)
+            ret = vy_ast.Decimal.from_node(node, value=val)
 
         if isinstance(input_type, IntegerT):
-            return vy_ast.Int.from_node(node, value=val)
+            ret = vy_ast.Int.from_node(node, value=val)
+
+        ret._metadata["type"] = input_type
```

Comment:
hmm right, we need to get better about propagating types during constant folding. this looks good, can you add a note to change this to `"known_type"` (per https://github.com/vyperlang/vyper/pull/3213)